homeassistant/components/myuplink/sensor.py

```diff
@@ -41,9 +41,48 @@
         state_class=SensorStateClass.MEASUREMENT,
         native_unit_of_measurement=UnitOfFrequency.HERTZ,
     ),
+    "43108": SensorEntityDescription(
+        key="fan_mode",
+        icon="mdi:fan",
+    ),
+    "43427": SensorEntityDescription(
+        key="status_compressor",
+        device_class=SensorDeviceClass.ENUM,
+        icon="mdi:heat-pump-outline",
+    ),
+    "49993": SensorEntityDescription(
+        key="elect_add",
+        device_class=SensorDeviceClass.ENUM,
+        icon="mdi:heat-wave",
+    ),
+    "49994": SensorEntityDescription(
+        key="priority",
+        device_class=SensorDeviceClass.ENUM,
+        icon="mdi:priority-high",
+    ),
```

Comment:
S series an F series have overlapping number series (at least with the old f series numbering) so this might not be enough. I kind of doubt that myuplink upgraded F series have adapted the numbering of the S series.

Ive not looked at the numbering for S series on myuplink uet  but it would sort of make more sense if matched modbus numbers, but maybe it doesnt?

Reply:
We are prepared for overlapping numbers. The numbers in the descriptor keys can be qualified with category name if needed. However I suspect that this can be a source of more work until everything calmes down...
I would really like to know how S-series presents itself in the myuplink API to try to find the most efficient solution.
The new firmware for my F730 was released about two weeks ago and i upgraded and migrated last week and quite a lot of sensors are missing in the public API. I noted that Nibe/myUplink have made some significant upgrades to the capabilities in the ios-app and on the web for F730 the last few days.

Reply:
So far as S series and F series never have same number for anything. So i think it should always be qualified.

If want to see the mostly complete databases, they are here https://github.com/yozik04/nibe/tree/master/nibe/data

We can probably ask some S series guys using the modbus setup test the myuplink integration of you dont have any such testers. Once the new release is public i suspect they will test it.

Reply:
@elupus when you mention S serie, i have a NIBE VVM S320 E EM 3x400V DK - what do you want me to test?
Without fully understand you point, i think it's fine to have these parameter_id config, and fallback to parameter_unit.

Reply:
@pajzo what does your parameter 43108 represent

Reply:
Or 49994? Or some other parameter for 'priority' you have

Reply:
> @pajzo what does your parameter 43108 represent

It is "Current fan mode" in the F730, = [0, 1, 2, 3, 4] It is assigned % as unit which is incorrect.

49994 is labelled Priority and can be Off, Hot water, Heating, Pool and so on.

From the metadata in each device-point I think we can make an educated guess what entity platform that is suitable. E.g. a parameter that can be zero or one and is writable should be represented by a switch. If it is read-only, a binary sensor would be appropriate.

Reply:
Im asking for S series report from @pajzo . As i said their number series are very likely overlapping and with completely different meaning for same number.

Reply:
Here is a output for my S320 - seems like the parameter_id's is unique.

```json
[
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "4",
        "parameterName": "Current outdoor temper­ature (BT1)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:53:53+00:00",
        "value": 0.4,
        "strVal": "0.4°C",
        "smartHomeCategories": [
            "sh-outdoorTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "8",
        "parameterName": "Supply line (BT2)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:54:54+00:00",
        "value": 38.4,
        "strVal": "38.4°C",
        "smartHomeCategories": [
            "sh-supplyTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "10",
        "parameterName": "Return line (BT3)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:54:38+00:00",
        "value": 28.2,
        "strVal": "28.2°C",
        "smartHomeCategories": [
            "sh-returnTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "11",
        "parameterName": "Hot water top (BT7)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:46:44+00:00",
        "value": 48.8,
        "strVal": "48.8°C",
        "smartHomeCategories": [
            "sh-hwTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "12",
        "parameterName": "Hot water char­ging (BT6)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:47:25+00:00",
        "value": 46.0,
        "strVal": "46°C",
        "smartHomeCategories": [
            "sh-hwTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "54",
        "parameterName": "Average temper­ature (BT1)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-08T22:58:57+00:00",
        "value": -0.6,
        "strVal": "-0.6°C",
        "smartHomeCategories": [
            "sh-outdoorTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "781",
        "parameterName": "Degree minutes",
        "parameterUnit": "DM",
        "writable": true,
        "timestamp": "2024-02-09T03:54:58+00:00",
        "value": -796.0,
        "strVal": "-796DM",
        "smartHomeCategories": [],
        "minValue": -30000.0,
        "maxValue": 30000.0,
        "stepValue": 20.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "1708",
        "parameterName": "Calcu­lated supply climate system 1",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:40:12+00:00",
        "value": 35.0,
        "strVal": "35°C",
        "smartHomeCategories": [
            "sh-supplyTemp"
        ],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "Varmepumpe 1",
        "parameterId": "2491",
        "parameterName": "Return line (EB101-BT3)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:55:09+00:00",
        "value": 27.9,
        "strVal": "27.9°C",
        "smartHomeCategories": [],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "Varmepumpe 1",
        "parameterId": "2494",
        "parameterName": "Supply line (EB101-BT12)",
        "parameterUnit": "°C",
        "writable": false,
        "timestamp": "2024-02-09T03:53:54+00:00",
        "value": 36.8,
        "strVal": "36.8°C",
        "smartHomeCategories": [],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [],
        "scaleValue": "0.1",
        "zoneId": null
    },
    {
        "category": "NIBE VVM S320 E EM 3x400V DK",
        "parameterId": "10895",
        "parameterName": "Pump: Heating medium (GP6)",
        "parameterUnit": "",
        "writable": false,
        "timestamp": "2024-02-08T23:45:49+00:00",
        "value": 0.0,
        "strVal": "0",
        "smartHomeCategories": [],
        "minValue": null,
        "maxValue": null,
        "stepValue": 1.0,
        "enumValues": [
            {
                "value": "0",
                "text": "0",
                "icon": ""
            },
            {
                "value": "1",
                "text": "1",
                "icon": ""
            }
        ],
        "scaleValue": "1",
        "zoneId": null
    }
]
```